src/concrete/ml/onnx/onnx_impl_utils.py

```diff
@@ -223,3 +226,38 @@ def onnx_avgpool_compute_norm_const(
         norm_const = numpy.prod(kernel_shape)
 
     return norm_const
+
+
+def rounded_comparison(
+    x: numpy.ndarray, y: numpy.ndarray, lsbs_to_remove: int, operation: ComparisonOperationType
+) -> Tuple[bool]:
+    """Comparison operation using `round_bit_pattern` function.
+
+    `round_bit_pattern` rounds the bit pattern of an integer.
+    It also checks for any potential overflow. If so, it readjusts the LSBs accordingly.
+
+    The parameter `lsbs_to_remove` can either be an integer specifying the number of LSBS to
+    remove, or an `AutoRounder` object that determines the required number of LSBs based on the
+    specified number of MSBs to retain. But in our case, we choose to compute the LSBs manually.
+
+    Args:
+        x (numpy.ndarray): Input tensor
+        y (numpy.ndarray): Input tensor
+        lsbs_to_remove (int): The number of the least significant bits to remove
+        operation (ComparisonOperationType): Comparison operation
+
+    Returns:
+        Tuple[bool]: If x and y satisfy the comparison operator.
+    """
+
+    assert isinstance(lsbs_to_remove, int)
+
+    # In this context, `round_bit_pattern` is used as a truncate operation.
+    # Consequently, we subtract a term, called `half` that will subsequently be re-added during the
+    # `round_bit_pattern` process.
+    half = 1 << (lsbs_to_remove - 1)
```

Comment:
is this a workaround ? if so could you add a FIXME with an issue here to indicate that it should be removed once we have the truncate feature ?

Reply:
I added a fixme in the top of the function, specifying the changes that have to be made when truncate is released